### ISSUE TITLE
Spurious CA1305 for Guid.Parse after targeting net7.0

### ISSUE THREAD

**Issue by bgrainger** [dotnet/runtime#69356]
### Description

After updating the `TargetFramework` for an existing program to `net7.0`, analysis starts generating spurious CA1305 warnings for `Guid.Parse`, because it now has an overload that takes `IFormatProvider` (but completely ignores it) https://github.com/dotnet/runtime/blob/af2a48ddaec9a3e27002b0d3b80c07431c2e38c1/src/libraries/System.Private.CoreLib/src/System/Guid.cs#L1490

If `TreatWarningsAsErrors` is `true`, the program no longer compiles.

### Reproduction Steps

Create a new Console app with this csproj:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
    <AnalysisLevel>latest-all</AnalysisLevel>
    <NoWarn>CA1014</NoWarn>
  </PropertyGroup>

</Project>
```

And this source code:

```csharp
Console.WriteLine(Guid.Parse("11223344-5566-7788-99AA-BBCCDDEEFF00"));
```

Run `dotnet run`:

> C:\Code\Projects\GuidParse\Program.cs(1,19): error CA1305: The behavior of 'Guid.Parse(string)' could vary based on the current user's locale settings. Replace this call in '<top-level-statements-entry-point>' with a call to 'Guid.Parse(string, IFormatProvider)'. [C:\Code\Projects\GuidParse\GuidParse.csproj]

If it's changed back to `<TargetFramework>net6.0</TargetFramework>`, the program compiles and runs normally.

### Expected behavior

No CA1305 analysis warning is introduced for existing working code when updating the target framework to `net7.0`, particularly because no benefit is gained by using the suggested overload that takes an `IFormatProvider`.

### Actual behavior

`dotnet run` produces this output and doesn't run the program:

> > C:\Code\Projects\GuidParse\Program.cs(1,19): error CA1305: The behavior of 'Guid.Parse(string)' could vary based on the current user's locale settings. Replace this call in '<top-level-statements-entry-point>' with a call to 'Guid.Parse(string, IFormatProvider)'. [C:\Code\Projects\GuidParse\GuidParse.csproj]

### Regression?

Yes; no problem with this code targeting `net6.0`.

### Configuration

7.0.100-preview.4.22252.9
Windows 10, x64

**Comment by bgrainger** [comment]
Could be a duplicate of https://github.com/dotnet/roslyn-analyzers/issues/5988, but affects the SDK since analyzers ship "in the box".

**Comment by stephentoub** [comment]
The default level for CA1305 is "silent"; even with warnings-as-errors, it shouldn't be causing the build to fail.  Do you override that level for CA1305 to make it a warning?

**Comment by bgrainger** [comment]
Yes, I did add `<AnalysisLevel>latest-all</AnalysisLevel>` to the default project generated by `dotnet new console`. (That's the setting I use in a different project where I first encountered this issue.)

It does _not_ fail with `<AnalysisLevel>latest</AnalysisLevel>`. However, it does generate the warning with `<AnalysisLevel>latest-recommended</AnalysisLevel>`, which seems like it could be a problem?

None of this changes the fact that the warning is spurious at _any_ warning level. It is simply not true that "The behavior of 'Guid.Parse(string)' could vary based on the current user's locale settings."

**Comment by stephentoub** [comment]
> None of this changes the fact that the warning is spurious at any warning level.

While that's true for Guid, I don't think it's desirable for the analyzer to maintain a blocklist of types that ignore the argument.

FWIW, the main reason the analyzer defaults to silent / hidden is _because_ it can have false positives.  Using latest-all to upgrade every analyzer to warnings is basically accepting that possibility.
